subsys/bluetooth/controller/ll_sw/ull_sync_iso.c

```diff
@@ -207,126 +226,385 @@ int ull_sync_iso_reset(void)
 	return 0;
 }
 
-struct ll_sync_iso *ull_sync_iso_get(uint8_t handle)
+struct ll_sync_iso_set *ull_sync_iso_get(uint8_t handle)
 {
 	if (handle >= CONFIG_BT_CTLR_SCAN_SYNC_ISO_SET) {
 		return NULL;
 	}
 
-	return &ll_sync_iso_pool[handle];
+	return &ll_sync_iso[handle];
 }
 
-uint8_t ull_sync_iso_handle_get(struct ll_sync_iso *sync)
+uint8_t ull_sync_iso_handle_get(struct ll_sync_iso_set *sync)
 {
-	return mem_index_get(sync, ll_sync_iso_pool,
-			     sizeof(struct ll_sync_iso));
+	return mem_index_get(sync, ll_sync_iso,
+			     sizeof(struct ll_sync_iso_set));
 }
 
 uint8_t ull_sync_iso_lll_handle_get(struct lll_sync_iso *lll)
 {
 	return ull_sync_handle_get(HDR_LLL2ULL(lll));
 }
 
-void ull_sync_iso_release(struct ll_sync_iso *sync_iso)
+void ull_sync_iso_release(struct ll_sync_iso_set *sync_iso)
 {
 	mem_release(sync_iso, &sync_iso_free);
 }
 
-void ull_sync_iso_setup(struct ll_sync_iso *sync_iso,
+void ull_sync_iso_setup(struct ll_sync_iso_set *sync_iso,
 			struct node_rx_hdr *node_rx,
-			struct pdu_big_info *big_info)
+			uint8_t *acad, uint8_t acad_len)
 {
 	uint32_t ticks_slot_overhead;
-	struct node_rx_sync_iso *se;
+	uint32_t sync_iso_offset_us;
 	uint32_t ticks_slot_offset;
 	struct lll_sync_iso *lll;
 	struct node_rx_ftr *ftr;
-	uint32_t sync_offset_us;
-	struct node_rx_pdu *rx;
+	struct pdu_big_info *bi;
+	uint32_t ready_delay_us;
 	uint32_t interval_us;
-	uint16_t interval;
-	uint16_t handle;
+	struct pdu_adv *pdu;
+	uint8_t bi_size;
+	uint8_t handle;
 	uint32_t ret;
+	uint8_t sca;
+
+	if (acad_len < (PDU_BIG_INFO_CLEARTEXT_SIZE +
+			PDU_ADV_DATA_HEADER_SIZE)) {
+		return;
+	}
+
+	/* FIXME: Parse and find the BIGInfo */
+	bi_size = acad[PDU_ADV_DATA_HEADER_LEN_OFFSET];
+	bi = (void *)&acad[PDU_ADV_DATA_HEADER_DATA_OFFSET];
 
 	lll = &sync_iso->lll;
-	handle = ull_sync_iso_handle_get(sync_iso);
+	(void)memcpy(lll->seed_access_addr, &bi->seed_access_addr,
+		     sizeof(lll->seed_access_addr));
+	(void)memcpy(lll->base_crc_init, &bi->base_crc_init,
+		     sizeof(lll->base_crc_init));
+
+	(void)memcpy(lll->data_chan_map, bi->chm_phy,
+		     sizeof(lll->data_chan_map));
+	lll->data_chan_map[4] &= ~0xE0;
```

Comment:
```suggestion
	lll->data_chan_map[4] &= 0x1F;
```
Seems more common in the code, and should give the same result, right?

Reply:
Fixed.